### ISSUE TITLE
MS Exchange setup fails

### ISSUE THREAD

**Issue by HKowi** [jobisoft/EAS-4-TbSync#215]
## Your environment

TbSync version: 4.3
EAS-4-TbSync version: 4.1.5 (also with 4.1.4)
Thunderbird version: 102.7

## Expected behavior
Open the authentication windows for OAuth2

## Actual behavior
Windows opens but is empty.

## Steps to reproduce
Add Account -> Exchange -> O365 account: fill in UN and account name

To help resolving your issue, enable debug logging (TbSync Account Manager -> Help) and send me the debug.log via e-mail (use the title of your issue as subject of the email).
Log-File
** Wed Jan 25 2023 16:09:26 GMT+0000 (Greenwich Mean Time) **
[Querry EAS autodiscover V2 URL] : https://autodiscover-s.outlook.com/autodiscover/autodiscover.json?Email=<my_email_here>&Protocol=ActiveSync

**Comment by dbujold** [comment]
I have this issue as well. When authentication window pops out, it's trying to connect to server "undefined".

TbSync version: 4.3
EAS-4-TbSync version: 4.1.5
Thunderbird version: 102.7.1

**Comment by jobisoft** [comment]
There is a bug in Thunderbird and I think they are fixing it with 102.7.2. I will check back next week.

**Comment by dbujold** [comment]
Thank you!

**Comment by ccox-tlit** [comment]
Just resolved the Thuderbrid OAUTH bug and can finally get O365 mail.  
Decided to try exchange cal sync. 
Getting a blank page when it tries to start the OAUTH login flow.   The popup has this url
https://undefined/?response_type=code&client_id=undefined&redirect_uri=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fnativeclient&scope=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fv2.0%2Fauthorize&login_hint=emailaddress

the one that points to MS I believe is coming from my changes within Enterprise Apps in Azure

**Comment by eddiermar** [comment]
> Just resolved the Thuderbrid OAUTH bug and can finally get O365 mail. Decided to try exchange cal sync. Getting a blank page when it tries to start the OAUTH login flow. The popup has this url https://undefined/?response_type=code&client_id=undefined&redirect_uri=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fnativeclient&scope=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fv2.0%2Fauthorize&login_hint=emailaddress
> 
> the one that points to MS I believe is coming from my changes within Enterprise Apps in Azure

Hi there... I am facing the same problem... could you explain the fix you made?

**Comment by fdellwing** [comment]
I can confirm this problem with TB 102.7.1 (build 2) x64 on Windows 10.

It seems, that this code here no longer works how it is supposed to: https://github.com/jobisoft/EAS-4-TbSync/blob/70f288f0098b710beb3c8a5f3aa8652b229a76a1/content/includes/network.js#L57-LL94

**Comment by dundee111** [comment]
> I have this issue as well. When authentication window pops out, it's trying to connect to server "undefined".
> 
> TbSync version: 4.3 EAS-4-TbSync version: 4.1.5 Thunderbird version: 102.7.1

I've had the same issue here.

**Comment by ccox-tlit** [comment]
> > Just resolved the Thuderbrid OAUTH bug and can finally get O365 mail. Decided to try exchange cal sync. Getting a blank page when it tries to start the OAUTH login flow. The popup has this url https://undefined/?response_type=code&client_id=undefined&redirect_uri=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fnativeclient&scope=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fv2.0%2Fauthorize&login_hint=emailaddress
> > the one that points to MS I believe is coming from my changes within Enterprise Apps in Azure
> 
> Hi there... I am facing the same problem... could you explain the fix you made?

Do you mean for thunderbird? I can't fix tbsync.
for thunderbird not connecting to O365 - you will need to be a GLobal Admin that can consent to apps
I just got it working in Ubuntu 22, here are my steps. 
I created a new clean VM 
Did  apt-get update/upgrade 
Try and use the builtin thunderbird - fails 
Download the tar directly from thunderbird 
Ran through the manual install steps from this support article 
[https://support.mozilla.org/en-US/kb/installing-thunderbird-linux](https://us-east-2.protection.sophos.com/?d=mozilla.org&u=aHR0cHM6Ly9zdXBwb3J0Lm1vemlsbGEub3JnL2VuLVVTL2tiL2luc3RhbGxpbmctdGh1bmRlcmJpcmQtbGludXg=&i=NjBmZWQ4NjA3OGEwMGMwZTdjYTg2MDdl&t=MEI5bnhUTkR4ZUVkT3RGVElvamhsVVFsdzhMMnc0Q0JGSXJiQXdTM3M3cz0=&h=357997e7e6704f3db6a915511aca34e5&s=AVNPUEhUT0NFTkNSWVBUSVaZGf97I2NZwZXJFhBEql3o-ZXu9W3kZYRcgTzg-HQ8FQ) 
It blew away my broken profile that wouldn’t connect and presented the first run screen. 
Fill in the info for external_data; MS popup for password entry. 
Soft fail – select manual configuration 
Change the password to oauth 
Redo login process – success mail begins downloading
Somewhere in the flow above it prompts for admin consent than gives you the option to login with your Global Admin account

**Comment by jasonhoran** [comment]
Same here.  The OAuth2 bug for Office365 Outlook has been resolved for mailboxes in TB 102.7.1 (release 2) that came out today.  I'm on Ubuntu 22.04.  I authorised correctly and my mailbox has synced fully without difficulty.
However TbSync doesn't work.  I could not connect to the server to synchronise calendar or contacts.  I tried to recreate the account and got this URL in the OAuth2 authorisation window which leads to a blank page:
`https://undefined/?response_type=code&client_id=undefined&redirect_uri=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fnativeclient&scope=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fv2.0%2Fauthorize&login_hint=redacted%40redacted.com`

**Comment by jobisoft** [comment]
[EAS-4-TbSync.zip](https://github.com/jobisoft/EAS-4-TbSync/files/10608481/EAS-4-TbSync.zip)

Fixed?

**Comment by jasonhoran** [comment]
> [EAS-4-TbSync.zip](https://github.com/jobisoft/EAS-4-TbSync/files/10608481/EAS-4-TbSync.zip)
> 
> Fixed?

Yes, it worked perfectly.  Thank you!

**Comment by fdellwing** [comment]
> [EAS-4-TbSync.zip](https://github.com/jobisoft/EAS-4-TbSync/files/10608481/EAS-4-TbSync.zip)
> 
> Fixed?

Works for me.

**Comment by dieterbeller** [comment]
Works for me too. Thanks for the quick fix!

**Comment by HKowi** [comment]
Works prefect again. Thanks for your help on this one.
Closing ticket.

**Comment by jobisoft** [comment]
Let's keep this open, until the new version has passed review and is shipped automatically to all users.

**Comment by HKowi** [comment]
Fair enough - no problem.

**Comment by smmehta1822** [comment]
Thank you for quick fix.

**Comment by stheine** [comment]
working fine. thanks!

**Comment by bvhoesel** [comment]
Yep working again. Now also via normal add-in update. Thanks al lot.

**Comment by fgo-stei** [comment]
> [EAS-4-TbSync.zip](https://github.com/jobisoft/EAS-4-TbSync/files/10608481/EAS-4-TbSync.zip)
> 
> Fixed?

Yes, it works for me too. Thanks for the quick fix.

**Comment by Giermann** [comment]
> Yep working again. Now also via normal add-in update. Thanks al lot.

I can confirm, the provieded Zip is working.
But I was unable to update to 4.2 via normal update - and also the release is not listed under "releases" in repository yet.

**Comment by ThomasStruller** [comment]
the EAS-4-TbSync.zip works perfeckt on win11 thunderbird 102.7.1 Thank you

**Comment by jobisoft** [comment]
The update has been reviewed and is now shipped to users automatically.

**Comment by fftmp** [comment]
> The update has been reviewed and is now shipped to users automatically.

Hello, sorry for posting in the closed issue, but this is related to new release, that was produced here.

Whether you will publish new releases on github?
Archlinux have package for this addon, but [it takes xpi ](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=thunderbird-eas-4-tbsync) from github, not from mozilla addons registry.
What source can be treated as "official"?

**Comment by jobisoft** [comment]
official is mozilla
